packages/web/src/components/track/OwnerActionButtons.tsx

```diff
@@ -0,0 +1,140 @@
+import { ReactNode, useCallback } from 'react'
+
+import { useGetPlaylistById, useGetTrackById } from '@audius/common/api'
+import { ID } from '@audius/common/models'
+import {
+  publishTrackConfirmationModalUIActions,
+  trackPageActions,
+  useEditTrackModal
+} from '@audius/common/store'
+import {
+  Flex,
+  IconButton,
+  IconPencil,
+  IconRocket,
+  IconShare
+} from '@audius/harmony'
+import { useDispatch } from 'react-redux'
+
+import Tooltip from 'components/tooltip/Tooltip'
+
+const { requestOpen: openPublishTrackConfirmationModal } =
+  publishTrackConfirmationModalUIActions
+const { makeTrackPublic } = trackPageActions
+
+const messages = {
+  share: 'Share',
+  edit: 'Edit Track',
+  publish: 'Make Public'
+}
+
+type OwnerActionButtonProps = {
+  contentId: ID
+  contentType: 'track' | 'collection'
+  isDisabled?: boolean
+  isLoading?: boolean
+  rightActions?: ReactNode
+  bottomBar?: ReactNode
+  isDarkMode?: boolean
+  isMatrixMode: boolean
+  showIconButtons?: boolean
+  onClickShare: (e?: any) => void
+}
+
+export const OwnerActionButtons = ({
+  contentId,
+  contentType,
+  isDisabled,
+  isLoading,
+  rightActions,
+  bottomBar,
+  showIconButtons,
+  onClickShare
+}: OwnerActionButtonProps) => {
+  const dispatch = useDispatch()
+  const { data: track } = useGetTrackById({ id: contentId })
+  const { data: collection } = useGetPlaylistById({ playlistId: contentId })
+  const isUnlisted =
+    contentType === 'track' ? track?.is_unlisted : collection?.is_private
+  const { onOpen: onEditTrackOpen } = useEditTrackModal()
+
+  const onStopPropagation = useCallback((e: any) => e.stopPropagation(), [])
+
+  // TODO: move this up a level
+  const handleEdit = useCallback(() => {
```

Comment:
tbh i'm ok with this living at this level. wanna move away from piping things down and this feels appropriately co-located. thoughts?

Reply:
valid, I was thinking more along the lines of consistency but I think you're right about stopping the drilling